### PR TITLE
docs: update CR harbor-cluster defined

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: zhuxiaoyang <sunzhu@yunify.com>

### CODE REVIEW

File: docs/cr_HarborCluster_spec.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  storage: 1Gi
+  replicas: 2
+  version: "12"
+
+# storage service configurations
+# might be external cloud storage services or
+# default in-cluster storage (minIO)
+storage:
+storage:
+  kind: minIO/filesystem/azure/gcs/s3 ....
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: kind for local using 'inCluster'

File: docs/cr_HarborCluster_spec.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  options:
+   minIO:
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]:
```yaml
  options:
    provider: minIO
    spec:
       aaa: 11
       bbb: 222
```

File: docs/cr_HarborCluster_spec.md (hunk 1 above)
Comment (steven-zou) [reply docs/cr_HarborCluster_spec.md]: Add docs to describe what does `inCluster` mean

File: docs/cr_HarborCluster_spec.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        		requests:
+          		storage: 10Gi
+      resources:
+        requests:
+          memory: 512Mi
+          cpu: 250m
+        limit:
+          memory: 512Mi
+          cpu: 250m
+   filesystem:
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: Remove `filesystem`

File: docs/cr_HarborCluster_spec.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    		metadata:
+    		  # if you want to reuse data, set the same name.
+      		name: data
+    		spec:
+      		accessModes:
+        		- ReadWriteOnce
+      		resources:
+        		requests:
+          		storage: 10Gi
+      resources:
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: add more comments if necessary, e.g.: optional specs

File: docs/cr_HarborCluster_spec.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+   	  # Supply number of replicas.
+  		# For standalone mode, supply 1. For distributed mode, supply 4 or more (should be even).
+  		# Note that the operator does not support upgrading from standalone to distributed mode.
+      replicas: 4
+      version: RELEASE.2020-01-03T19-12-21Z
+      # VolumeClaimTemplate allows a user to specify how volumes inside a MinIOInstance
+      volumeClaimTemplate:
+    		metadata:
+    		  # if you want to reuse data, set the same name.
+      		name: data
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: hide this attribute

File: docs/cr_HarborCluster_spec.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    		spec:
+      		accessModes:
+        		- ReadWriteOnce
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: expose `storage class`

File: docs/cr_HarborCluster_spec.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        requests:
+          memory: 512Mi
+          cpu: 250m
+        limit:
+          memory: 512Mi
+          cpu: 250m
+   filesystem:
+      rootdirectory: /storage
+      maxthreads: 100
+    azure:
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: Only leave one storage configuration as an example, comment others

File: docs/cr_HarborCluster_spec.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+      memory: 2048Mi
+    storage: 5Gi
+  sentinel:
+    replicas: 3
+
+# database service (PostgresSQL) configuration
+database:
+  storage: 1Gi
+  replicas: 2
+  version: "12"
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md]: add `storageClass`?

Comment (steven-zou) [reply docs/cr_HarborCluster_spec.md]: expose `resources:` for database too

File: docs/cr_HarborCluster_spec.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+priority: 30
+
+# pod instance number
+# required
+replicas: 3
+
+# source registry of images
+imageSource:
+  registry: harbor.com
```
Comment (soulseen) [inline docs/cr_HarborCluster_spec.md:35]: Is `registry` field is redundant? `imagePullSecrets` already contain the registry address.

File: docs/cr_HarborCluster_spec.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+# log configurations
+log:
+  level: debug
+  
+# The option to set repository read only.
+readOnly: false
+
+# set proxy
+proxy:
+  http_proxy: 10.10.20.2
```
Comment (soulseen) [inline docs/cr_HarborCluster_spec.md:48]: please use Camel-Case.

File: docs/cr_HarborCluster_spec.md (hunk 9 above)
Comment (steven-zou) [reply docs/cr_HarborCluster_spec.md:35]: I don't think so. This registry field is used for defining the image pulling path. Even the secret contains the related registry path, I don't think it's suitable to refer to the code.

File: docs/cr_HarborCluster_spec.md (hunk 10 above)
Comment (steven-zou) [reply docs/cr_HarborCluster_spec.md:48]: +1

File: docs/cr_HarborCluster_spec.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+  registry: harbor.com
+  imagePullSecrets:
+   - pSecret
+   
+# log configurations
+log:
+  level: debug
+  
+# The option to set repository read only.
+readOnly: false
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md:44]: Let's remove `readOnly` at present.

File: docs/cr_HarborCluster_spec.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+# required
+replicas: 3
+
+# source registry of images
+imageSource:
+  registry: harbor.com
+  imagePullSecrets:
+   - pSecret
+   
+# log configurations
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md:39]: Suspended until core operator supports those configurations items

File: docs/cr_HarborCluster_spec.md (hunk 9 above)
Comment (steven-zou) [reply docs/cr_HarborCluster_spec.md:35]: core operator:

```yaml
  components:
    harbor-core:
      image: <your_repo>/repo:tag
```

File: docs/cr_HarborCluster_spec.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+trivy:
+  github_token: 123
+
+# extra configuration options for chartmeseum
+chartMuseum:
+  absoluteURL: true
+  
+# cache service(Redis) configurations
+# required
+redis:
```
Comment (steven-zou) [inline docs/cr_HarborCluster_spec.md:78]: add kind for redis/database:
  ```yaml
   redis:
     kind: inCluster # external
  ```

Comment (steven-zou) [reply docs/cr_HarborCluster_spec.md:78]: For external, we only need a secret reference.

Document the secret content format to users.

**Comment by Ghostbaby** [comment]
LGTM

**Pull request merged**